### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#8219] [Releases]
## ✨ What’s New in v0.8.0? ✨

Hey everyone, we’re excited to announce the release of version 0.8.0! This update brings a mix of new features, enhancements, and crucial bug fixes. Here’s a quick rundown:

### 🔥 Key Feature

**Parallel Execution of Nodes in Workflows** by @takatost, @zxhlyh, and @YIXIAO0 in #8192.
   
![image](https://github.com/user-attachments/assets/98b209a1-d72e-4e8b-8bfe-7ca0babf27c4)

Nodes can now be executed in parallel within a workflow, greatly increasing the execution speed. This feature is especially beneficial for complex workflows that involve multiple steps or processes, allowing for quicker completion times and improved performance.

Dive deeper into the details and unleash the full potential of these new features by exploring our [latest blog post](https://dify.ai/blog/accelerating-workflow-processing-with-parallel-branch) and [documentation](https://docs.dify.ai/guides/workflow/orchestrate-node)!

### 🚀 New Features

- **Support `gpt-4o-2024-08-06` and `json_schema` for Azure OpenAI Service**: Support for the latest GPT-4o model and JSON schema for Azure OpenAI by @hjlarry in #7648
- **Support Oracle Cloud Infrastructure Generative AI Service**: Oracle Cloud Infrastructure is now a supported model provider by @tmuife in #7775
- **Support Fish Audio TTS**: Added support for Fish Audio Text-to-Speech models by @leng-yue in #7982

### ⚠️ Deprecated Features

- **Deprecate N to 1 Retrieval** by @zxhlyh in #8134
  The N-to-1 retrieval strategy is officially deprecated in this version, of which the entrance will be closed but applications that have selected this feature will still be retained. We recommend switching to the more flexible multi-path retrieval strategy to boost your application's retrieval efficiency.

### ⚙️ Enhancements

- **Update App Published Time After Clicking Publish Button**: The published time of an app now updates correctly when you click the publish button by @vicoooo26 in #7801

### 🛠️ Bug Fixes

- **Fix Notion Table Extract**: Fixed issues with extracting data from Notion tables by @JohnJyong in #7925
- **Fix Nvidia Rerank Top N Missed**: Addressed issues with Nvidia rerank top N functionality by @JohnJyong in #8185
- **Fix Claude Credential Validation**: Resolved credential validation issues for Claude by @crazywoola in #8109

That’s it for this release! As always, we appreciate your feedback and contributions. Do it for you! 🚀

---

### Docker compose deployments

> [!WARNING]
> The `docker-compose.yaml` has been refactored. If you've made any changes to the file, make sure to check out the "Upgrade to new docker compose deployment" section above for usage and migration tips.

1. Back up your customized docker-compose YAML file (optional)

   ```bash
   cd docker
   cp docker-compose.yaml docker-compose.yaml.$(date +%s).bak
   ```

2. Get the latest code from the main branch

   ```bash
   git checkout main
   git pull origin main
   ```

3. Stop the service，Command, please execute in the docker directory

   ```bash
   docker compose down
   ```

4. Back up data

   ```bash
   tar -cvf volumes-$(date +%s).tgz volumes
   ```

5. Upgrade services

   ```bash
   docker compose up -d
   ```

### Source Code deployments

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the main branch:

   ```bash
   git checkout main
   git pull origin main
   ```

3. Update Python dependencies:

   ```bash
   cd api
   poetry install
   ```

7. Then, let's run the migration script:

   ```bash
   poetry shell
   flask db upgrade
   ```

8. Finally, run API server, Worker and Web frontend Server again.

---

## What's Changed
* feat: support auto generate i18n translate by @iamjoel in https://github.com/langgenius/dify/pull/6964
* fix: frontend handle sometimes server not generate the wrong follow up data struct by @iamjoel in https://github.com/langgenius/dify/pull/7916
* improve the notion table extract by @JohnJyong in https://github.com/langgenius/dify/pull/7925
* Fix/segment create with api by @JohnJyong in https://github.com/langgenius/dify/pull/7928
* chore: #7603 i18n by @ZuzooVn in https://github.com/langgenius/dify/pull/7931
* fix rerank model value is empty string by @JohnJyong in https://github.com/langgenius/dify/pull/7937
* Chore/add default step 1x url by @crazywoola in https://github.com/langgenius/dify/pull/7933
* Chore/add roadmap by @crazywoola in https://github.com/langgenius/dify/pull/7943
* fix(api): Code-Based Extension cause error on position map sorting by @jim60105 in https://github.com/langgenius/dify/pull/7934
* chore: #7943 i18n by @ZuzooVn in https://github.com/langgenius/dify/pull/7948
* Revert "fix: an issue of keyword search feature in application log list" by @takatost in https://github.com/langgenius/dify/pull/7949
* feat: ops trace add llm model by @ZhouhaoJiang in https://github.com/langgenius/dify/pull/7306
* chore: fix inverted index japanese translation by @os1ma in https://github.com/langgenius/dify/pull/7957
* fix: split text keep separator by @Sumkor in https://github.com/langgenius/dify/pull/7930
* Feat: update app published time after clicking publish button by @vicoooo26 in https://github.com/langgenius/dify/pull/7801
* fix:empty voice occurs when xinference CosyVoice tts model by @leslie2046 in https://github.com/langgenius/dify/pull/7958
* fix: get commit sha failed in translate action by @iamjoel in https://github.com/langgenius/dify/pull/7959
* feat: add health check for pg and redis in docker-compose.middleware.yaml (#7961) by @hwzhuhao in https://github.com/langgenius/dify/pull/7962
* Feat:remove estimation of embedding cost by @JzoNgKVO in https://github.com/langgenius/dify/pull/7950
* chore: refactor the beichuan model by @hjlarry in https://github.com/langgenius/dify/pull/7953
* fix wrong error message by @JohnJyong in https://github.com/langgenius/dify/pull/7972
* chore: translate i18n files by @github-actions in https://github.com/langgenius/dify/pull/7965
* Concurrent calls to the Wenxin model, and the exception problem when obtaining the token is fixed by @wochuideng in https://github.com/langgenius/dify/pull/7976
* fix spliter length missed by @JohnJyong in https://github.com/langgenius/dify/pull/7987
* feat: add gpt-4o-2024-08-06 and json_schema for azure openAI service by @hjlarry in https://github.com/langgenius/dify/pull/7648
* feat: support fish audio TTS by @leng-yue in https://github.com/langgenius/dify/pull/7982
* feat: return page number of pdf documents upon retrieval by @jasonkang14 in https://github.com/langgenius/dify/pull/7749
* Fix/datasets api description error by @Designerxsh in https://github.com/langgenius/dify/pull/8025
* feat: add video、audio、abbr href message rendering and optimize image rendering by @charli117 in https://github.com/langgenius/dify/pull/6868
* Update milvus-standalone version and expose required ports for the container. by @winsonwhe in https://github.com/langgenius/dify/pull/7709
* support huawei cloud obs storage (#7980) by @hwzhuhao in https://github.com/langgenius/dify/pull/7981
* Add OCI(Oracle Cloud Infrastructure) Generative AI Service as a Model Provider by @tmuife in https://github.com/langgenius/dify/pull/7775
* fix the tooltip in tools node by @YIXIAO0 in https://github.com/langgenius/dify/pull/8055
* chore: #7348, support query conversations by updated_at by @ZuzooVn in https://github.com/langgenius/dify/pull/8047
* fix: change milvus init args from (host, port) to (url, token) by @zc277584121 in https://github.com/langgenius/dify/pull/8019
* ifEsle node add regex match by @charli117 in https://github.com/langgenius/dify/pull/8007
* If else add regexmatch by @crazywoola in https://github.com/langgenius/dify/pull/8059
* fix: parameter rule by @crazywoola in https://github.com/langgenius/dify/pull/8064
* fix: overflow issues in chat history by @YIXIAO0 in https://github.com/langgenius/dify/pull/8062
* Revert "fix: parameter rule" by @AAEE86 in https://github.com/langgenius/dify/pull/8070
* fix: i18n typo by @ZuzooVn in https://github.com/langgenius/dify/pull/8077
* fix: knowledge input by @crazywoola in https://github.com/langgenius/dify/pull/8065
* Fix variable typo by @ZuzooVn in https://github.com/langgenius/dify/pull/8084
* feat: support redis sentinel mode by @erigo in https://github.com/langgenius/dify/pull/7756
* fix LangSmith project config error by @charli117 in https://github.com/langgenius/dify/pull/7996
* fix: parameter input by @crazywoola in https://github.com/langgenius/dify/pull/8076
* feat:add fishaudio in xinference by @leslie2046 in https://github.com/langgenius/dify/pull/8100
* chore: keep dify compose file consistent format by @soulteary in https://github.com/langgenius/dify/pull/8102
* Translate billing to PT-BR by @thalessalazar in https://github.com/langgenius/dify/pull/8105
* fix: Claude can not validate credientials by @crazywoola in https://github.com/langgenius/dify/pull/8109
* feat: support more model types and builtin tools on aws/sagemaker by @ybalbert001 in https://github.com/langgenius/dify/pull/8061
* Fix: tongyi code wrapper works not stable by @zouchengzhuo in https://github.com/langgenius/dify/pull/7871
* add dify-sandbox health check in docker-compose.yaml (#8121) by @hwzhuhao in https://github.com/langgenius/dify/pull/8124
* fix: ollama text embedding 500 error by @crazywoola in https://github.com/langgenius/dify/pull/8131
* chore: offline n to 1 retrieval by @zxhlyh in https://github.com/langgenius/dify/pull/8134
* Modify model parameters in Spark LLMs and zhipuai LLMs by @AAEE86 in https://github.com/langgenius/dify/pull/8078
* fix: user-select style and pre-create iframe in embed.js by @cr-zhichen in https://github.com/langgenius/dify/pull/8093
* chore: translate i18n files by @github-actions in https://github.com/langgenius/dify/pull/8135
* fix: html raw render by @crazywoola in https://github.com/langgenius/dify/pull/8138
* Update LICENSE - remove 'SaaS' from restriction term definition by @guchenhe in https://github.com/langgenius/dify/pull/8143
* update qwen-long max token by @crazywoola in https://github.com/langgenius/dify/pull/8157
* Fix variable typo (cont) by @ZuzooVn in https://github.com/langgenius/dify/pull/8161
* feat(api/core/model_runtime/entities/defaults.py): Add TOP_K in default parameters. by @laipz8200 in https://github.com/langgenius/dify/pull/8167
* chore: bump super-linter to v7 by @bowenliang123 in https://github.com/langgenius/dify/pull/8148
* add volcengine tos storage by @yaoice in https://github.com/langgenius/dify/pull/8164
* code merge error by @JohnJyong in https://github.com/langgenius/dify/pull/8183
* chore: upgrade expat from 2.6.2-2 to 2.6.3-1 by @ZuzooVn in https://github.com/langgenius/dify/pull/8182
* nvidia rerank top n missed by @JohnJyong in https://github.com/langgenius/dify/pull/8185
* fix: The length of the tag should between 1 and 50 (#8187) by @hwzhuhao in https://github.com/langgenius/dify/pull/8188
* fix(api/core/rag/extractor/excel_extractor.py): filter excel empty sheet by @JohnJyong in https://github.com/langgenius/dify/pull/8194
* refactor(api/core): Improve type hints and apply ruff formatter in agent runner and model manager. by @laipz8200 in https://github.com/langgenius/dify/pull/8166
* docs: replace docker-compose with docker compose by @crazywoola in https://github.com/langgenius/dify/pull/8195
* feat: Parallel Execution of Nodes in Workflows by @takatost in https://github.com/langgenius/dify/pull/8192
* chore: translate i18n files by @github-actions in https://github.com/langgenius/dify/pull/8202
* chore(api/core): apply ruff reformatting by @bowenliang123 in https://github.com/langgenius/dify/pull/7624
* fix: input chat input wrong padding by @iamjoel in https://github.com/langgenius/dify/pull/8207
* chore(api/models): apply ruff reformatting by @bowenliang123 in https://github.com/langgenius/dify/pull/7600
* fix the tooltip in tool nodes by @YIXIAO0 in https://github.com/langgenius/dify/pull/8215
* revert page column by @JohnJyong in https://github.com/langgenius/dify/pull/8217
* fix(docker/docker-compose.yaml): Set default value for `REDIS_SENTINEL_SOCKET_TIMEOUT` and `CELERY_SENTINEL_SOCKET_TIMEOUT` by @laipz8200 in https://github.com/langgenius/dify/pull/8218

## New Contributors
* @jim60105 made their first contribution in https://github.com/langgenius/dify/pull/7934
* @os1ma made their first contribution in https://github.com/langgenius/dify/pull/7957
* @Sumkor made their first contribution in https://github.com/langgenius/dify/pull/7930
* @github-actions made their first contribution in https://github.com/langgenius/dify/pull/7965
* @wochuideng made their first contribution in https://github.com/langgenius/dify/pull/7976
* @leng-yue made their first contribution in https://github.com/langgenius/dify/pull/7982
* @Designerxsh made their first contribution in https://github.com/langgenius/dify/pull/8025
* @winsonwhe made their first contribution in https://github.com/langgenius/dify/pull/7709
* @zc277584121 made their first contribution in https://github.com/langgenius/dify/pull/8019
* @AAEE86 made their first contribution in https://github.com/langgenius/dify/pull/8070
* @thalessalazar made their first contribution in https://github.com/langgenius/dify/pull/8105
* @zouchengzhuo made their first contribution in https://github.com/langgenius/dify/pull/7871
* @cr-zhichen made their first contribution in https://github.com/langgenius/dify/pull/8093
* @yaoice made their first contribution in https://github.com/langgenius/dify/pull/8164

**Full Changelog**: https://github.com/langgenius/dify/compare/0.7.3...0.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.8.0'>v0.8.0</a>.</em>

**Comment by sanbei101** [comment]
伟大的进步꒰ *•ɷ•* ꒱

**Comment by PNjobs** [comment]
Is the parallel flow only for chatflow? I upgraded and I dont see it in workflow

**Reply by laipz8200** [reply]
It is supported in both workflow and chatflow.
